src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -81,17 +73,28 @@ export const openFolder = folderId => {
       // PB: Pouch Mango queries don't return the total count...
       // and so the fetchMore button would not be displayed unless... see FileList
       const folder = await getAdapter(getState()).getFolder(folderId)
-      return dispatch({
-        type: OPEN_FOLDER_SUCCESS,
-        folder,
-        fileCount: folder.contents.meta.count || 0,
-        files: folder.contents.data
-      })
+      /*
+        Since getFolder is async, if we have a very bad network we can receive multiple response 
+        if the user clicks multiple times. We are not sure about the order of the response. 
+        So before dispatching the success, we check if the folder is still the one we want to open 
+        and display. To do that, we get the folder id from the latest OPEN_FOLDER call
+      */
+      const currentFolderId = getOpenedFolderId(getState())
+      if (folder._id === currentFolderId) {
+        return dispatch({
+          type: openFolderActionSucess,
+          folder,
+          fileCount: folder.contents.meta.count || 0,
+          files: folder.contents.data
+        })
```

Comment:
IMHO this kind of thing should be done in the reducer. It should ignore the success if it is for the wrong id.

Reply:
You absolutely right! But we have a "special" reducer in drive. We splited / sliced our reducer and state. So for instance here https://github.com/cozy/cozy-drive/blob/master/src/drive/web/modules/navigation/duck/reducer.js#L197 I can't have a direct access to the "global" state since in this method I'm scoped. So no acess `state.openedFolderId` :(

Maybe I missed something to get access to it. I'm very open to suggestion here !